.github/actions/cl2-modules/egw/modules/masq-metrics.yaml

```diff
@@ -1,48 +1,81 @@
         threshold: 1
-      - name: EGW Masquerade Delay - 95th Percentile
+      - name: P95
         query: quantile(0.95, egw_scale_test_masquerade_delay_seconds_total{k8s_instance="{{ .instance }}"})
         threshold: 2
-      - name: EGW Masquerade Delay - 99th Percentile
+      - name: P99
         query: quantile(0.99, egw_scale_test_masquerade_delay_seconds_total{k8s_instance="{{ .instance }}"})
         threshold: 2
 
-  - Identifier: EGWLeakedPingsTotal{{ .metricsSuffix }}
+  - Identifier: MasqueradeDelayLeakedPingsTotal{{ .metricsSuffix }}
     Method: GenericPrometheusQuery
     Params:
       action: {{ .action }}
-      metricName: "EGW Leaked Pings Total ({{ .metricsSuffix }})"
+      metricName: Masquerade Delay {{ .metricsSuffix }} - Leaked Pings Total
       metricVersion: v1
       unit: count
+      enableViolations: true
       queries:
-      - name: EGW Leaked Pings - Total
+      - name: Total
         query: sum(egw_scale_test_leaked_requests_total{k8s_instance="{{ .instance }}"})
 
-  - Identifier: EGWPodCountMetrics{{ .metricsSuffix }}
+  - Identifier: MasqueradeDelayPodCount{{ .metricsSuffix }}
     Method: GenericPrometheusQuery
     Params:
       action: {{ .action }}
-      metricName: "EGW Pod Count Metrics ({{ .metricsSuffix }})"
+      metricName: Masquerade Delay {{ .metricsSuffix }} - Pod Count
       metricVersion: v1
       unit: pod
+      enableViolations: true
       queries:
-      - name: EGW Total Number of Client Pods
+      - name: Total
         query: count(egw_scale_test_failed_tests_total{k8s_instance="{{ .instance }}"})
-      - name: EGW Total Number of Failed Client Pods
+        threshold: {{ .replicas }}
+        lowerBound: true
+      - name: Failed
         query: sum(egw_scale_test_failed_tests_total{k8s_instance="{{ .instance }}"})
         threshold: 0
+
+  - Identifier: MasqueradeDelayCiliumCPUUsage{{ .metricsSuffix }}
+    Method: GenericPrometheusQuery
+    Params:
+      action: {{ .action }}
+      metricName: Masquerade Delay {{ .metricsSuffix }} - Cilium CPU Usage
+      metricVersion: v1
+      unit: cpu
+      enableViolations: true
+      queries:
+      - name: P50
+        query: quantile(0.50, avg_over_time(rate(cilium_process_cpu_seconds_total[1m])[%v:10s]))
+      - name: Max
+        query: max(avg_over_time(rate(cilium_process_cpu_seconds_total[1m])[%v:10s]))
+
+  - Identifier: MasqueradeDelayCiliumMemUsage{{ .metricsSuffix }}
+    Method: GenericPrometheusQuery
+    Params:
+      action: {{ .action }}
+      metricName: Masquerade Delay {{ .metricsSuffix }} - Cilium Memory Usage
+      metricVersion: v1
+      unit: MB
+      enableViolations: true
+      queries:
+      - name: P50
+        query: quantile(0.5, max_over_time(cilium_process_resident_memory_bytes[%v]) / 1024 / 1024)
+      - name: Max
+        query: max(max_over_time(cilium_process_resident_memory_bytes[%v]) / 1024 / 1024)
```

Comment:
nit: do you plan to add cpu/memory thresholds?
and similar for other metrics with `enableViolations: true`

Reply:
Makes sense, added. I've been a bit more generous compared to the previously configured ones, because the scraping is now limited to a shorter period of time in which we perform quite a few operations. Let's see how it goes, we can always tune it afterwards (or make it configurable) if need arises. 

As for the others, I'm not currently planning on adding an explicit threshold (it is a bit tricky with the stress ones, as it depends on the specific test combination, as well as on whether the EGW is enabled or not), but I've enabled violations regardless as I always forget, and then get confused about why thresholds don't work as expected.